campaignion_tracking/campaignion_tracking.module

```diff
@@ -184,13 +184,13 @@ function campaignion_tracking_webform_redirect_alter($redirect, Submission $subm
  */
 function _campaignion_tracking_snippets() {
   $snippets = module_invoke_all('campaignion_tracking_snippets');
-  drupal_alter('campaignion_tracking_snippets', $snippets);
   foreach ($snippets as &$snippet) {
     if (!is_array($snippet)) {
       $snippet = ['js' => $snippet];
     }
     $snippet += ['event' => 'tracking-allowed'];
   }
+  drupal_alter('campaignion_tracking_snippets', $snippets);
```

Comment:
Is that a benefit? The difference I see is that other modules now need to provide their additional snippets as array, including an event name, while before just the snippet would have been enough and the event name optional. The old version seemed safer to me. Am I overlooking anything else here? :thinking:

Reply:
Jep. The original hook is called before the normalization. So if you just *provide* snippets you can rely on the normalization. If on the other hand you want to manipulate snippets defined elsewhere you can rely on a canonical format, because the alter hook is called after the normalization.

Reply:
I mean:

* If you want to define snippets: Use `hook_campaignion_tracking_snippets()` and rely on the normalization.
* If you want to change snippets: Use `hook_campaignion_tracking_snippets_alter()` and rely on getting the normalized snippets as input for altering.

Reply:
Makes sense :+1: